rune/libenclave/internal/runtime/pal/skeleton/liberpal-skeleton.c

```diff
@@ -487,6 +490,60 @@ int __pal_exec(char *path, char *argv[], pal_stdio_fds *stdio, int *exit_code)
 	return 0;
 }
 
+int __pal_create_process(pal_create_process_args *args)
+{
+	int pid;
+	if ((pid = fork()) < 0)
+		return -1;
+	else if (pid == 0) {
+		int exit_code, ret;
+
+		ret = __pal_exec_v1(args->path, args->argv, args->stdio, &exit_code);
+		exit(ret ? ret : exit_code);
+	} else
+		*args->pid = pid;
+
+	return 0;
+}
+
+int __pal_exec_v2(pal_exec_args *attr)
+{
+        int status;
+        waitpid(attr->pid, &status, 0);
+
+        if (WIFEXITED(status) || WIFSIGNALED(status))
+                *attr->exit_value = WEXITSTATUS(status);
+
+        return 0;
+}
+
+int __pal_get_local_report(void *targetinfo, void *report)
+{
+	uint8_t report_data[64] = { 0, };
+	struct sgx_report report_align;
+	int ret;
+
+	ret = SGX_ENTER_3_ARGS(ECALL_REPORT, (void *)secs.base, targetinfo,
+						report_data, &report_align);
+	if (ret) {
+		fprintf(stderr, "failed to get report\n");
+		return ret;
+	}
+
+	memcpy(report, &report_align, SGX_REPORT_SIZE);
+	if(is_debug_mode) {
+		printf("ok to get report\n");
```

Comment:
You need to use the stdio fds to print the debugging message instead of using printf() directly. This means stdio fds set by __pal_init() should be globally stored in liberal-skeleton.c.

Reply:
> You need to use the stdio fds to print the debugging message instead of using printf() directly. This means stdio fds set by __pal_init() should be globally stored in liberal-skeleton.c.

The stdio fds isn't set by __pal_init() but __pal_exec(). In the case of  __pal_get_local_report is called at bootstrap, __pal_exec is not called by `rune`, so fds isn't initialized.